### PR TITLE
unified community page - READY FOR REVIEW #2

### PR DESCRIPTION
**Pull request by Seroxdesign**
## Overview

**What features/fixes does this PR include?**

- Unify community pages:

- [x] Patrons
- [x] Players
- [x] Guilds
- [ ] Elders (Skipping over due to lack of data/components)

Mockups:

https://www.figma.com/file/FxPTQ45mdnvZitp9jmIUQ3/Unified-Community-page?type=design&node-id=701-2024&mode=design

**Please provide the GitHub issue number**

Closes #

https://www.notion.so/meta-game/Improving-UX-02ea24e2fdc14cc4be0bebbede3b9487?p=223b75c206874b2d9e8d2cd9e86e452a&pm=s

image
## Assets

[Include screenshots/videos if it makes reviewing easier.]

https://github.com/MetaFam/TheGame/assets/69639595/1ea60c19-f5c5-4c9b-b5f1-9b1f945a7b91

### CODE REVIEW

File: packages/web/pages/community/index.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            <Tab
+              _selected={{
+                color: 'teal.200',
+                borderBottom: '2px solid #81E6D9',
+              }}
+              w="100%"
+            >
+              Guilds
+            </Tab>
+            <Tab
```
Comment (alalonde) [inline packages/web/pages/community/index.tsx:76]: lot of copy-paste here

**Comment by alalonde** [comment]
Do we still want the landing page menu to link to `/players`? I would think this would include a redirect to /community. Also, each tab IMO should correspond to its own URL so as not to break back / forward navigation

**Review by alalonde — CHANGES_REQUESTED** [review]

**Comment by Seroxdesign** [comment]
> Do we still want the landing page menu to link to `/players`? I would think this would include a redirect to /community. Also, each tab IMO should correspond to its own URL so as not to break back / forward navigation, AND so we can link to each tab individually

I can change the landing page to redirect to the community, I suppose I can also make a URL Param for players/guilds/patrons to keep the navigation stable. Thanks for revuew

File: packages/web/pages/community/index.tsx (hunk 1 above)
Comment (Seroxdesign) [reply packages/web/pages/community/index.tsx:76]: Understood will clean up

File: packages/web/pages/community/index.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  useEffect(() => {
+    const { query } = router;
+    if (query.tab) {
+      const index = communityTabs.findIndex(tab => tab.link === query.tab);
+      setActiveTab({ link: query.tab as string, index });
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [router.query]);
+
+  const handleTabChange = (index: number) => {
```
Comment (alalonde) [inline packages/web/pages/community/index.tsx:67]: This sort of thing should be in a `useCallback`

**Review by alalonde — APPROVED** [review]

**Pull request merged**